oximeter/impl/src/schema/mod.rs

```diff
@@ -153,18 +159,58 @@ fn validate_timeseries_name(s: &str) -> Result<&str, MetricsError> {
     }
 }
 
+/// Text descriptions for the target and metric of a timeseries.
+#[derive(Clone, Debug, Default, Deserialize, JsonSchema, Serialize)]
+pub struct TimeseriesDescription {
+    pub target: String,
+    pub metric: String,
+}
+
+/// Measurement units for timeseries samples.
+#[derive(Clone, Copy, Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
+#[serde(rename_all = "snake_case")]
+// TODO-completeness: Include more units, such as power / temperature.
+// TODO-completeness: Decide whether and how to handle dimensional analysis
+// during queries, if needed.
+pub enum Units {
+    Count,
+    Bytes,
+}
+
+impl Units {
+    /// Construct a `Units::Count`.
+    pub const fn count() -> Self {
```

Comment:
I'm not sure this method is worth it - `Units::count()` is two characters longer than just `Units::Count`

Reply:
I need it for the `#[serde(default = "...")]` attribute, which requires a callable function rather than a variant name, unfortunately. I agree, it's silly.

Reply:
Removed in 5610bb467